File: Job.ts
export interface Job {
  id: number;
  title: string;
  company: string;
  companyLogo?: string;
  location: string;
  salary: string;
  experience: string;
  jobType: JobType;
  category?: string;
  description: string;
  responsibilities: string[];
  requirements: string[];
  skills: string[];
  benefits?: string[];
  vacancies?: number;
  postedDate: string;
  lastDate?: string;
  status?: JobStatus;
  isFeatured?: boolean;
  isRemote?: boolean;
  isSaved?: boolean;
  isApplied?: boolean;
}

export interface JobApplication {
  id: number;
  jobId: number;
  userId: number;
  appliedDate: string;
  status: ApplicationStatus;
  resumeUrl?: string;
  coverLetter?: string;
}

export interface JobFilter {
  keyword?: string;
  location?: string;
  category?: string;
  experience?: string;
  jobType?: JobType;
  salaryMin?: number;
  salaryMax?: number;
  page?: number;
  limit?: number;
  sortBy?: "latest" | "salary" | "company";
}

export interface JobSearchResponse {
  jobs: Job[];
  total: number;
  page: number;
  totalPages: number;
}

export interface SavedJob {
  id: number;
  jobId: number;
  savedAt: string;
}

export type JobType =
  | "Full Time"
  | "Part Time"
  | "Internship"
  | "Remote"
  | "Contract"
  | "Freelance";

export type JobStatus =
  | "Open"
  | "Closed"
  | "Expired";

export type ApplicationStatus =
  | "Applied"
  | "Under Review"
  | "Interview"
  | "Selected"
  | "Rejected";

File: Job.py
from typing import Optional
from datetime import datetime

from pydantic import BaseModel, Field


class JobBase(BaseModel):
    company_id: int
    title: str = Field(..., min_length=3, max_length=150)
    description: str
    requirements: Optional[str] = None
    responsibilities: Optional[str] = None
    location: Optional[str] = None
    employment_type: str = "Full Time"
    experience_level: Optional[str] = None
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    skills: Optional[str] = None
    vacancies: int = 1
    status: str = "Open"
    application_deadline: Optional[datetime] = None


class JobCreate(JobBase):
    pass


class JobUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    requirements: Optional[str] = None
    responsibilities: Optional[str] = None
    location: Optional[str] = None
    employment_type: Optional[str] = None
    experience_level: Optional[str] = None
    salary_min: Optional[float] = None
    salary_max: Optional[float] = None
    skills: Optional[str] = None
    vacancies: Optional[int] = None
    status: Optional[str] = None
    application_deadline: Optional[datetime] = None


class JobResponse(JobBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True